daemon/container_operations_unix.go

```diff
@@ -175,6 +176,41 @@ func (daemon *Daemon) setupIpcDirs(c *container.Container) error {
 	return nil
 }
 
+func (daemon *Daemon) setupSecretDir(c *container.Container) error {
```

Comment:
shouldn't this func cleanup on error? I mean like unmount and remove secrets dir.